### ISSUE TITLE
Error building Metacall formula

### ISSUE THREAD

**Issue by FirePing32** [metacall/homebrew#4]
I ran `./build.sh` to build the Formula for Metacall, but it spits out some cmake error after building node_loader.
```bash
[ 88%] Built target java_loader
[ 88%] Linking CXX shared module ../../../libext_loader.so
cd /tmp/metacall-20231231-67744-bz44a6/core-0.5.27/build/source/loaders/ext_loader && /usr/local/Cellar/cmake/3.28.1/bin/cmake -E cmake_link_script CMakeFiles/ext_loader.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -bundle -Wl,-headerpad_max_install_names -o ../../../libext_loader.so CMakeFiles/ext_loader.dir/source/ext_loader.c.o CMakeFiles/ext_loader.dir/source/ext_loader_impl.cpp.o  -Wl,-rpath,/tmp/metacall-20231231-67744-bz44a6/core-0.5.27/build ../../../libmetacall.dylib -pthread
[ 88%] Linking CXX shared module ../../../libnode_loader.so
cd /tmp/metacall-20231231-67744-bz44a6/core-0.5.27/build/source/loaders/node_loader && /usr/local/Cellar/cmake/3.28.1/bin/cmake -E cmake_link_script CMakeFiles/node_loader.dir/link.txt --verbose=1
[ 88%] Built target ext_loader
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -bundle -Wl,-headerpad_max_install_names -o ../../../libnode_loader.so CMakeFiles/node_loader.dir/source/node_loader.c.o CMakeFiles/node_loader.dir/source/node_loader_impl.cpp.o CMakeFiles/node_loader.dir/source/node_loader_port.cpp.o CMakeFiles/node_loader.dir/source/node_loader_trampoline.cpp.o  -Wl,-rpath,/tmp/metacall-20231231-67744-bz44a6/core-0.5.27/build -Wl,-rpath,/usr/local/Cellar/metacall/0.5.27/lib ../../../libmetacall.dylib /usr/local/Cellar/metacall/0.5.27/lib/libnode.83.dylib -pthread
[ 88%] Built target node_loader
make: *** [all] Error 2

/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/git --version
/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/curl --version
/usr/local/Homebrew/Library/Homebrew/ignorable.rb:29:in `block in raise'
BuildError: Failed executing: cmake --build . --target install
1. raise
2. ignore
3. backtrace
4. irb
5. shell
```

Platform: Ventura 13.5
Arch: AMD64/Intel

**Comment by FirePing32** [comment]
Attaching logs

[Complete terminal output](https://github.com/metacall/homebrew/files/13829203/build-logs.txt)
[01.cmake](https://github.com/metacall/homebrew/files/13829218/01.cmake.txt)
[02.cmake](https://github.com/metacall/homebrew/files/13829225/02.cmake.txt)

**Comment by FirePing32** [comment]
Cmake is unable to find Python headers, although they are packed in the python installation provided by Homebrew. Instead of modifying the [CMakeList](https://github.com/metacall/core/blob/develop/CMakeLists.txt) we can pass include flags to clang/cmake at runtime.

**Comment by FirePing32** [comment]
I guess some extra `cmake` flags need to be added to https://github.com/metacall/core to include header file directories, as it can be different on each system. 
cc @AkechiShiro

**Comment by AkechiShiro** [comment]
I'm sorry @FirePing32 but I don't have my dev setup anymore and I'm too busy to get time to investigate the issue at the moment, since it is related to Python headers, the build fails most likely due to this : 
```
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -O3 -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -bundle 
-Wl,-headerpad_max_install_names -o ../../../libfile_loader.so CMakeFiles/file_loader.dir/source/file_loader.c.o CMakeFiles/file_loader.dir/source
/file_loader_impl.c.o  -Wl,-rpath,/tmp/metacall-20240104-37483-rwhozr/core-0.5.27/build ../../../libmetacall.dylib -pthread 
In file included from /tmp/metacall-20240104-37483-rwhozr/core-0.5.27/source/loaders/py_loader/source/py_loader.c:24:
/tmp/metacall-20240104-37483-rwhozr/core-0.5.27/source/loaders/py_loader/include/py_loader/py_loader_impl.h:30:10: fatal error: 'Python.h' file no
t found
#include <Python.h>
         ^~~~~~~~~~
[ 88%] Linking CXX shared module ../../../libmock_loader.so
1 error generated.
cd /tmp/metacall-20240104-37483-rwhozr/core-0.5.27/build/source/loaders/mock_loader && /usr/local/Cellar/cmake/3.28.1/bin/cmake -E cmake_link_scri
pt CMakeFiles/mock_loader.dir/link.txt --verbose=1
make[2]: *** [source/loaders/py_loader/CMakeFiles/py_loader.dir/source/py_loader.c.o] Error 1
make[1]: *** [source/loaders/py_loader/CMakeFiles/py_loader.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Could you clarify what do you mean different on each system (MacOS vs Linux or is it a difference between MacOS versions that is causing the failure?) 

You mean that there needs specific cmake flags for compiling under MacOS ?

**Comment by FirePing32** [comment]
I mean the user may have headers at multiple locations. Ex - An anaconda environment, Code headers, or the default headers provided by the Python installation.

**Comment by FirePing32** [comment]
@viferga Will create a PR soon.

**Comment by viferga** [comment]
@FirePing32 Cool.

**Comment by FirePing32** [comment]
@viferga I tried the above approach, even `cmake` uses that extra flag, but it is still failing at finding `Python.h` (similar error like above trace).

**Comment by FirePing32** [comment]
Update: If you see line 1005, it is failing at building `plugin_extension`, not exactly the `py_loader`. 
![Screenshot 2024-01-26 at 2 49 26 AM](https://github.com/metacall/homebrew/assets/43035576/bedb0a27-edec-4392-b542-dcdda1f52d83)
[This](https://github.com/metacall/core/blob/b0d8bc9a977b71031003383b2556c48f1214aed1/source/extensions/plugin_extension/source/plugin_extension.cpp#L126) line tries to import the loaders and fails to find `Python.h` when importing `py_loader`. Do we also need to pass Python's include path while building `plugin_extension` ?

**Comment by viferga** [comment]
Strange... you are defining properly the paths, can you verify "Python.h" is available here: `/usr/local/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12`  in your filesystem?

Now it seems to find properly the paths:
```
-- Found Python3: /usr/local/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12 (found version "3.12.1") found components: Development Development.Module Development.Embed
```
